ambry-api/src/main/java/com/github/ambry/named/NamedBlobRecord.java

```diff
@@ -87,6 +105,20 @@ public NamedBlobRecord(String accountName, String containerName, String blobName
     this(accountName, containerName, blobName, blobId, expirationTimeMs, version, 0);
   }
 
+  /**
+   * @param accountName the account name.
+   * @param containerName the container name.
+   * @param blobName the blob name within the container.
+   * @param blobId the blob ID for the blob content in ambry storage.
+   * @param expirationTimeMs the expiration time in milliseconds since epoch, or -1 if the blob should be permanent.
+   * @param version the version of this named blob.
+   * @param digest the digest of this blob
+   */
+  public NamedBlobRecord(String accountName, String containerName, String blobName, String blobId,
```

Comment:
I feel like we have too many constructors which might not nessessary, we can try to combine them.  we can have a follow up pr to do it.